client/data/emails/use-get-email-accounts-query.ts

```diff
@@ -13,8 +19,12 @@ export const getCacheKey = ( siteId, domain ) => [ 'emailAccounts', siteId, doma
  * @returns {data, error, isLoading} Returns and object with the
  * data associated to the SiteId & Domain
  */
-export const useEmailAccountsQuery = ( siteId, domain, queryOptions = {} ) => {
-	return useQuery(
+export const useGetEmailAccountsQuery = (
+	siteId: number,
+	domain: string,
+	queryOptions?: UseQueryParams[ 2 ]
```

Comment:
Because `useQuery` uses function overloading, there are multiple signatures for the function. The reason I use `UseQueryParams[ 2 ]` rather than `UseQueryOptions` is that when we pass 3 arguments to `useQuery`, `UseQueryOptions` isn't the actual type being used. Instead, `Omit<UseQueryOptions, 'queryKey' | 'queryFn'>` is used. To defer that rather complex type definition to `react-query`, I opted for this approach.